lib/active_stash/model_reflection.rb

```diff
@@ -0,0 +1,40 @@
+module ActiveStash
+  module ModelReflection
+
+    def self.fields(model)
+      fields = model.attribute_types.inject({}) do |attrs, (k,v)|
+        type = v.type
+
+        # ActiveRecord encryption is available from Rails 7.
+        if Rails::VERSION::MAJOR >= 7
+          type = ActiveRecord::Encryption::EncryptedAttributeType === v ? v.cast_type.type : v.type
+        end
+
+        attrs[k] = type
+        attrs
+      end
+
+      remove_lockbox_fields(model, fields)
+      remove_id_fields(fields)
```

Comment:
It took me a moment to work out what this is doing. Now I can see that its a refactoring of the original code but maybe we should add a comment explaining why this is "removing" fields?